src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int GROW_STEP = 1;
+    private int size;
+    private Object[] elements;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (checkSize()) {
```

Comment:
` if (size == array.length) { grow(); }`  this code have to be checkSize() method

Reply:
Maybe, they should be separated, bc 1 method - 1 responsibility?
I guess, they do different tasks.